pdns/syncres.cc

```diff
@@ -515,16 +536,18 @@ int SyncRes::asyncresolveWrapper(const ComboAddress& ip, bool ednsMANDATORY, con
      If '3', send bare queries
   */
 
-  SyncRes::EDNSStatus* ednsstatus;
-  ednsstatus = &t_sstorage.ednsstatus[ip]; // does this include port? YES
+  SyncRes::ednsstatus_t::iterator ednsstatus = t_sstorage.ednsstatus.find(ip); // does this include port? YES
+  ednsstatus = t_sstorage.ednsstatus.insert(ip).first;
```

Comment:
I'm not sure why this calls `find()` then always `insert()`?

Reply:
Ah, I started with find and then thought: insert does what I want, but some old code remained.